libs/bArray.lua

```diff
@@ -28,16 +28,26 @@ end
 	@name new
 	@desc Creates a new instance of a Byte Array. Alias: `byteArray()`.
 	@param stack?<table> An array of bytes.
+	@param pos?<int> A pointer to the current position in the stack when reading.
 	@returns byteArray The new Byte Array object.
 	@struct {
 		stack = { } -- The bytes stack
 	}
 ]]
 byteArray.new = function(self, stack)
 	return setmetatable({
+		pos = 1,
 		stack = (stack or { }) -- Array of bytes
 	}, self)
 end
+--[[@
+	@name bytesLeft
+	@desc Returns the number of bytes left in the stack.
+	@returns int Number of bytes left in the stack.
+]]
+byteArray.bytesLeft = function(self)
```

Comment:
Can you check the develop branch of Transfromage? I made something that is better than bytesLeft... PR it to your repo